### ISSUE TITLE
Steam causes Xorg to leak memory

### ISSUE THREAD

**Issue by kiroma** [ValveSoftware/steam-for-linux#5764]
#### Your system information

* Steam client version (build number or date): Sep 8 2018
* Distribution (e.g. Ubuntu): Kubuntu 18.04
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
When Steam is running for a prolonged amount of time it causes Xorg to leak memory.
This issue was primarily the root cause of #5279.

#### Steps for reproducing this issue:

1. Open steam
2. Wait
3. Check Xorg memory usage

**Comment by kiroma** [comment]
So I'm bumping this again since after a short session the X server managed to reach over 1GB of RAM usage, shortly before my computer froze

**Comment by A-UNDERSCORE-D** [comment]
re-Bumping this, issue persists

**Comment by A-UNDERSCORE-D** [comment]
Continues to be an issue.

**Comment by urmum-69** [comment]
Bumping.
I have no idea why such a serious problem is being utterly ignored here.

**Comment by A-UNDERSCORE-D** [comment]
Yeah I'll say "still an issue" here as well. Ive taken to keeping steam closed most of the time which really sucks for random game invites and such

**Comment by jilv220** [comment]
Same here. Still a issue. Leaving steam client open cause my xorg eating more and more memory.

**Comment by Ristovski** [comment]
Can confirm on Gentoo with XMonad - Steam just caused Xorg to accumulate almost a gigabyte of RAM after having Steam open for ~1 day. Closing Steam made Xorgs ram usage drop to a measly 132MB. 

Having multiple Steam sub-windows open (like friends list, chat etc) makes it even worse.

**Comment by CocaineJohnsson** [comment]
Steam still bloats xorg with gigabytes of unused pixmaps or windows over two and a half years later, often within the span of hours.

In this case steam has been running for 8 hours:
![image](https://user-images.githubusercontent.com/53689060/116609205-69010480-a934-11eb-8f81-3fb6fb69d293.png)


And here is the exact same process list after restarting steam and waiting for it to load
![image](https://user-images.githubusercontent.com/53689060/116609380-a06fb100-a934-11eb-9e83-a95c41dcbfc3.png)


Not ice the drastically lower Xorg memory use

**Comment by zergling-man** [comment]
Solution: Use Goldberg steam emulator :^)

**Comment by dakrk** [comment]
Happy late 3rd birthday to this issue 🎉

**Comment by ForteDexe** [comment]
the problem still exist, manjaro gnome 40

**Comment by nonchip** [comment]
aaaaaaand now it's leaking so many sockets i can't actually start **any**thing anymore (not even e.g. `glxgears` from outside of steam) after 5ish minutes of steam starting, due to `Maximum number of clients reached`...
